src/ppx/scene/scene_gltf_loader.cpp

```diff
@@ -0,0 +1,2966 @@
+                case cgltf_component_type_r_16u : return grfx::FORMAT_R16G16_UINT;
+                case cgltf_component_type_r_32u : return grfx::FORMAT_R32G32_UINT;
+                case cgltf_component_type_r_32f : return grfx::FORMAT_R32G32_FLOAT;
+            }
+        } break;
+
+        case cgltf_type_vec3: {
+            switch (pGltfAccessor->component_type) {
+                default: return grfx::FORMAT_UNDEFINED;
+                case cgltf_component_type_r_8   : return grfx::FORMAT_R8G8B8_SINT;
+                case cgltf_component_type_r_8u  : return grfx::FORMAT_R8G8B8_UINT;
+                case cgltf_component_type_r_16  : return grfx::FORMAT_R16G16B16_SINT;
+                case cgltf_component_type_r_16u : return grfx::FORMAT_R16G16B16_UINT;
+                case cgltf_component_type_r_32u : return grfx::FORMAT_R32G32B32_UINT;
+                case cgltf_component_type_r_32f : return grfx::FORMAT_R32G32B32_FLOAT;
+            }
+        } break;
+
+        case cgltf_type_vec4: {
+            switch (pGltfAccessor->component_type) {
+                default: return grfx::FORMAT_UNDEFINED;
+                case cgltf_component_type_r_8   : return grfx::FORMAT_R8G8B8A8_SINT;
+                case cgltf_component_type_r_8u  : return grfx::FORMAT_R8G8B8A8_UINT;
+                case cgltf_component_type_r_16  : return grfx::FORMAT_R16G16B16A16_SINT;
+                case cgltf_component_type_r_16u : return grfx::FORMAT_R16G16B16A16_UINT;
+                case cgltf_component_type_r_32u : return grfx::FORMAT_R32G32B32A32_UINT;
+                case cgltf_component_type_r_32f : return grfx::FORMAT_R32G32B32A32_FLOAT;
+            }
+        } break;
+    }
+    // clang-format on
+
+    return grfx::FORMAT_UNDEFINED;
+}
+
+static scene::NodeType GetNodeType(const cgltf_node* pGltfNode)
+{
+    if (IsNull(pGltfNode)) {
+        return scene::NODE_TYPE_UNSUPPORTED;
+    }
+
+    if (!IsNull(pGltfNode->mesh)) {
+        return scene::NODE_TYPE_MESH;
+    }
+    else if (!IsNull(pGltfNode->camera)) {
+        return scene::NODE_TYPE_CAMERA;
+    }
+    else if (!IsNull(pGltfNode->light)) {
+        return scene::NODE_TYPE_LIGHT;
+    }
+    else if (!IsNull(pGltfNode->skin) || !IsNull(pGltfNode->weights)) {
+        return scene::NODE_TYPE_UNSUPPORTED;
+    }
+
+    return scene::NODE_TYPE_TRANSFORM;
+}
+
+static grfx::SamplerAddressMode ToSamplerAddressMode(cgltf_int mode)
+{
+    switch (mode) {
+        default: break;
+        case GLTF_TEXTURE_WRAP_CLAMP_TO_EDGE: return grfx::SAMPLER_ADDRESS_MODE_CLAMP_TO_EDGE;
+        case GLTF_TEXTURE_WRAP_MIRRORED_REPEAT: return grfx::SAMPLER_ADDRESS_MODE_MIRRORED_REPEAT;
+    }
+    return grfx::SAMPLER_ADDRESS_MODE_REPEAT;
+}
+
+template <typename GltfObjectT>
+static bool HasExtension(
+    const std::string& extensionName,
+    const GltfObjectT* pGltfObject)
+{
+    if (extensionName.empty() || IsNull(pGltfObject) || IsNull(pGltfObject->extensions)) {
```

Comment:
I'm not sure if I'm reading this the same way - right after this line is a loop that uses extensions_count and searches through the array for a specific string?

Reply:
I think the function is safe, that's why @Keenuts said it is not a big issue, but since extensions_count is a more well defined variable, it might be more proper to rely on that one instead of the array for early return.